test/grpc/route_guide/list-features.cc

```diff
@@ -0,0 +1,90 @@
+#include <algorithm>
+#include <string>
+
+#include "eventuals/finally.h"
+#include "eventuals/foreach.h"
+#include "eventuals/head.h"
+#include "eventuals/let.h"
+#include "eventuals/promisify.h"
+#include "eventuals/then.h"
+#include "gtest/gtest.h"
+#include "test/grpc/route_guide/route-guide-test.h"
+
+using grpc::Status;
+
+using eventuals::Finally;
+using eventuals::operator*;
+using eventuals::Foreach;
+using eventuals::Head;
+using eventuals::Let;
+using eventuals::Then;
+using routeguide::Feature;
+
+using eventuals::grpc::Stream;
+
+TEST_F(RouteGuideTest, ListFeaturesTest) {
+  const int32_t left = -750000000;
+  const int32_t right = -730000000;
+  const int32_t top = 420000000;
+  const int32_t bottom = 400000000;
+
+  std::vector<Feature> filtered;
+  size_t current = 0;
+
+  std::copy_if(
+      feature_list.begin(),
+      feature_list.end(),
+      std::back_inserter(filtered),
+      [&](const auto& feature) {
+        return feature.location().longitude() >= left
+            && feature.location().longitude() <= right
+            && feature.location().latitude() >= bottom
+            && feature.location().latitude() <= top;
+      });
+
+  routeguide::Rectangle rect;
+  rect.mutable_lo()->set_latitude(bottom);
+  rect.mutable_lo()->set_longitude(left);
+  rect.mutable_hi()->set_latitude(top);
+  rect.mutable_hi()->set_longitude(right);
+
+  auto e = [&]() {
+    return client().ListFeatures()
+        >> Then(Let([&](auto& call) {
+             return call.Writer().WriteLast(std::move(rect))
+                 >> Foreach(
+                        call.Reader().Read(),
+                        [&](Feature&& feature) {
+                          auto latitude = feature.location().latitude();
+                          auto longitude = feature.location().longitude();
+
+                          auto& expected_feature =
+                              filtered[current++];
+
+                          EXPECT_EQ(
+                              expected_feature.location().latitude(),
+                              latitude);
+
+                          EXPECT_EQ(
+                              expected_feature.location().longitude(),
+                              longitude);
+
+                          EXPECT_EQ(
+                              expected_feature.name(),
+                              feature.name());
+                        })
+                 >> Finally([&call](auto) {
+                      return call.Finish();
+                    })
+                 >> Then([](Status&& status) {
```

Comment:
Let's spell this completely to make it clear where it's coming from:
```suggestion
                 >> Then([](::grpc::Status&& status) {
```
And don't forget to remove the `using` above!